adapter-guide/develop-stix-adapter.md

```diff
@@ -0,0 +1,602 @@
+# Developing a new STIX-shifter adapter
+
+* [Introduction](stix-adapter-overview.md)
+* [Scenario](#scenario)
+* [Prerequisites](#prerequisites)
+* [Steps](#steps)
+
+## Scenario
+{: #scenario}
+
+### Participants
+This scenario involves a software developer (*Developer A*) and an end user (*User A*). *Developer A* wants to implement a new adapter for the STIX-shifter project that can support a particular security product (*Product A*). *User A* is another developer that uses the STIX-shifter library.
+
+### Problem to solve
+*User A* performs security monitoring with *Product A* and several other security products. The other products already have existing STIX-shifter adapters. 
+
+*User A* would like to:
+1.	Submit one STIX pattern to query all the user’s security products at once. The use of a STIX pattern simplifies the search process because *User A* does not need to know the query language or API calls for each security product. 
+1.	See the query results from all the security products in one unified format (STIX bundle). With the assumption that the submitted pattern represents a potential security incident, the STIX bundle presents the query results in the context of the security event. 
+
+By implementing a new adapter, *Developer A* allows *Product A* to fit into the workflow.  
+
+
+## Prerequisites for *Developer A*
+{: #prerequisites}
+
+* Your development environment must use Python 3.6.
+* You must have access to the target data source. In the sample scenario, you must have access to Product A data source.
+* You must be familiar with Product A's query language and APIs.
+* You must be familiar or understand the following concepts:
+    * Observable objects. See [STIX™ Version 2.0. Part 4: Cyber Observable Objects](http://docs.oasis-open.org/cti/stix/v2.0/stix-v2.0-part4-cyber-observable-objects.html)
+    * Stix patterning. See [STIX™ Version 2.0. Part 5: STIX Patterning](https://docs.oasis-open.org/cti/stix/v2.0/stix-v2.0-part5-stix-patterning.html)
+
+
+## Steps
+{: #steps}
+
+1. To develop a STIX-shifter adapter for a data source:
+    1. Fork the `IBM/stix-shifter` repository from https://github.com/IBM/stix-shifter to work on your own copy of the library.
+    1. [Create a Translation module](#translation-mod). 
+    1. [Create a Transmission module](#transmission-mod).
+    1. Create a pull request to merge your changes in the `IBM/stix-shifter` repository.
+
+1. To use the newly developed STIX-shifter adapter:
+    * Follow the steps in the "How to use" section of the `stix-shifter/README.md` file, in https://github.com/IBM/stix-shifter.
+
+### Create a Translation module
+{: #translation-mod}
+
+1. [Create a translation module folder](#translation-1)
```

Comment:
None of these links seem to work.

Reply:
Noted Danny. I'll make the corrections for all of the links.